CMakeLists.txt

```diff
@@ -456,7 +456,8 @@ include_directories(
     pal
     pal/inc
     pal/inc/rt
-    ${ICU_INCLUDE_PATH}
+    ${ICU_INCLUDE_PATH}/common
```

Comment:
why this is needed?

Reply:
Because @dilijev is working on things right now that have headers in <icu>/source/i18n in addition to source/common. Its easier if we normalize everything off of just icu/source

Reply:
Well, this is not a trivial change. Better if we keep it as is. There is a reason they kept the header files in two sep. folders. We shouldn't normalize them.

Reply:
Not sure what you mean. `ICU_INCLUDE_PATH` will have a known directory tree with files in known places. The dependency on ICU headers is effectively the directory that needs to be specified. In my WIP*, I'm adding `${ICU_INCLUDE_PATH}/i18n` because that's where those headers are.

* see: https://github.com/dilijev/ChakraCore/tree/intl-numfmt and this commit in particular:
https://github.com/dilijev/ChakraCore/commit/a04cde6ce70c1a9d92f2add6dc55b51a19a98cb4

https://github.com/dilijev/ChakraCore/blob/a04cde6ce70c1a9d92f2add6dc55b51a19a98cb4/lib/Runtime/Base/Chakra.Runtime.Base.vcxproj#L31-L32

If not this strategy, what would you propose? Have the build system specify two directories that are within the same parent folder for semantically the same dependency?

Reply:
1 - This is the place where you set the search path for compiler. If you put both `i18n` and `common` separately, you would normalize all the header files under.

How it's different than copying all the header files from those folders into a common folder?
Well, worst case, it might break and we won't even notice.

What's wrong with `#include <i18n/THIS_FILE>` and `#include <common/THAT_FILE>` ?

2 - ICU is optional. There is a reason we created a variable for `ICU_INCLUDE_PATH`... Once `${ICU_INCLUDE_PATH}` is not defined, what exactly do expect from compiler for `${ICU_INCLUDE_PATH}/common` or `${ICU_INCLUDE_PATH}/....` ??

If we didn't do this for Windows project files, well, it's time we make it right?

Reply:
I see your point. FWIW from what I've seen, including them this way seems to be common practice, and in the documentation is even shown as just the filename itself, implying that they would go even further to specify all of:

* `${ICU_INCLUDE_PATH}/common/` (has not been applicable in the project yet but is supported by the current scheme)
* `${ICU_INCLUDE_PATH}/common/unicode`
* `${ICU_INCLUDE_PATH}/i18n/unicode`

However, I understand where you are coming from, and agree there's no reason to keep adding more directories when we can specify the top level one instead.

This would mean that we need to update all of the ICU includes to start scoped to the true ICU directory.

Jack, do you think we should do that now or make a revision to this in a subsequent pull request? Up to you based on whether you feel like landing this immediately is important and an update can come later.

Reply:
Alternatively, only if INTL is enabled, add `${ICU_INCLUDE_PATH}../i18/`.. Mind `../` because `ICU_INCLUDE_PATH` == `common` already.

However, we should track from CMAKE and make sure INTL is enabled before adding that.

So, we won't be updating build scripts command lines (it's not just us!) and make the header sharing limited to INTL usage. I guess, this is what you want?

Reply:
I think that both of the following are fine, all things being considered equal: 

1. put the paths in the #includes
```CMake
include_directories(
  ...
  ${ICU_INCLUDE_PATH}
  )
```
```C
#include "common/unicode/uloc.h"
#include "i18n/header.h"
```
2. Scope the include
```CMake
if(SET ${ICU_INCLUDE_PATH})
  include_directories(
    ${ICU_INCLUDE_PATH}/common
    ${ICU_INCLUDE_PATH/i18n
    )
endif()
```
The latter takes fewer changes, so I would probably prefer the former at least for now.

Reply:
Note that ICU_INCLUDE_PATH will only be set when we are not using the system ICU -- it will be worth looking at where the i18n headers are when theyre installed on the system and probably choose whatever is necessary to conform to that

Reply:
> The latter takes fewer changes, so I would probably prefer the former at least for now.

Prefer the former because it takes more changes?

Reply:
I have no idea what I was on about there, I prefer solution #2.